File: VecMathREPLCpp/vecmathlib/Joke.cpp
#include "Joke.h"
#include <iostream>

Joke::Joke(std::string line)
{
	m_Lines.push_back(line);
}

Joke::Joke(std::string line1, std::string line2)
{
	m_Lines.push_back(line1);
	m_Lines.push_back(line2);
}

void Joke::printJoke(const Console& console) const
{
	if (!m_Lines.empty()) {
		console.Print(Console::VMF_BRIGHTGREEN, m_Lines[0]);
		console.NewLine();
	}
	if (m_Lines.size() >= 2) {
		console.Print(Console::VMF_BRIGHTCYAN, m_Lines[1]);
		console.NewLine();
	}
}

File: VecMathREPLJava/src/main/java/dae/gd/vecmathproject/Joke.java
package dae.gd.vecmathproject;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Koen.Samyn
 */
public class Joke {

    private List<String> lines = new ArrayList<>();

    public Joke(String line) {
        lines.add(line);
    }

    public Joke(String line1, String line2) {
        lines.add(line1);
        lines.add(line2);
    }

    void printJoke(VecMathListener listener) {
        if (lines.size() >= 1) {
            listener.printInfo(lines.get(0), true);
        }
        if (lines.size() >= 2) {
            listener.printInfo(lines.get(1), true);
        }
    }

}
